src/libstd/sys/windows/io.rs

```diff
@@ -48,6 +60,18 @@ impl<'a> IoSliceMut<'a> {
         }
     }
 
+    #[inline]
+    pub fn advance(&mut self, n: usize) {
+        if (self.vec.len as usize) < n {
+            panic!("advancing IoSliceMut beyond its length");
+        }
+
+        unsafe {
```

Comment:
Could we always uphold the minimum standard of having comments on all `unsafe {` blocks?
I think that's something you should be able to expect of something so central as the standard library.